src/libutil/experimental-features.cc

```diff
@@ -214,6 +214,22 @@ constexpr std::array<ExperimentalFeatureDetails, 13> xpFeatureDetails = {{
                 derivations that are themselves derivations outputs.
         )",
     },
+    {
+        .tag = Xp::ReadOnlyLocalStore,
+        .name = "read-only-local-store",
+        .description = R"(
+            Allow the use of the `read-only` parameter in local store URIs.
+
+            Set this parameter to `true` to allow stores with databases on read-only
+            filesystems to be opened for querying; ordinarily Nix will refuse to do this.
+
+            Enabling this setting disables the locking required for safe concurrent
+            access, so you should be certain that the database will not be changed.
```

Comment:
Why is this even? Would be great if we could capture this qualitatively, for example (made up):

```suggestion
            Enabling this setting disables the locking required for safe concurrent
            access to greatly increase performance, so you should be certain that the database will not be changed.
```